runtime/oti/j9.h

```diff
@@ -337,6 +337,7 @@ static const struct { \
  * Disable flattening of volatile field that is > 8 bytes for now, as the current implementation of copyObjectFields() will tear this field.
  */
 #define J9_IS_FIELD_FLATTENED(fieldClazz, romFieldShape) \
+		J9_ARE_NO_BITS_SET((romFieldShape)->modifiers, J9FieldFlagIsNullRestricted) || \
```

Comment:
> Also this [line](https://github.com/eclipse-openj9/openj9/blob/cda91b3765d104393feeac72d461a1200f3828e7/runtime/vm/createramclass.cpp#L2319) should be removed as it will always be false with the new javac. (If there are tests failed due to this remval, the test should be either rewritten or disabled)

I added a comment [here](https://github.com/eclipse-openj9/openj9/issues/18157#issuecomment-1729833947) asking about `J9ROMCLASS_IS_PRIMITIVE_VALUE_TYPE()`.